gwcs/wcs.py

```diff
@@ -332,7 +388,502 @@ def _invert(self, *args, **kwargs):
+            self._calc_approx_inv(max_inv_pix_error=5, inv_degree=None)
+
+        if self._approx_inverse is None:
+            if self.bounding_box is None:
+                x0 = np.ones(self.pixel_n_dim)
+            else:
+                x0 = np.mean(self.bounding_box, axis=-1)
+
+        if arg_dim == 0:
+            argsi = args
+
+            if nargs == 2 and self._approx_inverse is not None:
+                x0 = self._approx_inverse(*argsi)
+                if not np.all(np.isfinite(x0)):
+                    return [np.array(np.nan) for _ in range(nargs)]
+
+            result = optimize.fixed_point(f, x0, **kwargs)
+            return list(map(np.array, result))
+
+        else:
+            arg_shape = args_shape[1:]
+            nelem = np.prod(arg_shape)
+            inv = np.empty((nargs, nelem))
+
+            args = np.reshape(args, (nargs, nelem))
+
+            failed_result = np.full(nargs, np.nan)
+
+            for k in range(nelem):
+                argsi = args[:, k]
+
+                if nargs == 2 and self._approx_inverse is not None:
+                    x0 = self._approx_inverse(*argsi)
+                    if not np.all(np.isfinite(x0)):
+                        inv[:, k] = failed_result
+                        continue
+
+                result = optimize.fixed_point(f, x0, **kwargs)
+                inv[:, k] = result
+
+            return list(np.reshape(inv, args_shape))
+
+    def _invert_fp(self, *args, **kwargs):
+        """
+        Implements vectorized iterative inverse using fixed-point iterations.
+        """
+        try:
+            from scipy import optimize
+        except ImportError:
+            raise ImportError("Computation of the iterative inverse requires 'scipy'")
+
+        # remove optional 'invert'-specific keywords:
+        kwargs = dict(kwargs)
+        kwargs.pop('with_bounding_box', 0)
+        kwargs.pop('fill_value', 0)
+        kwargs['method'] = 'iteration'
+
+        tolerance = kwargs.pop('tolerance', 1e-4)
+        maxiter = kwargs.pop('maxiter', 40)
+        adaptive = kwargs.pop('adaptive', True)
+        detect_divergence = kwargs.pop('detect_divergence', True)
+        quiet = kwargs.pop('quiet', True)
+
+        args_shape = np.shape(args)
+        nargs = args_shape[0]
+        arg_dim = len(args_shape) - 1
+
+        if nargs != self.world_n_dim:
+            raise ValueError("Number of input coordinates is different from "
+                             "the number of defined world coordinates in the "
+                             f"WCS ({self.world_n_dim:d})")
+
+        if self.world_n_dim != self.pixel_n_dim:
+            raise NotImplementedError(
+                "Support for iterative inverse for transformations with "
+                "different number of inputs and outputs was not implemented."
+            )
+
+        # form equation:
+        def f(x):
+            return np.add(np.subtract(self.__call__(*x), argsi), x)
```

Comment:
> Wouldn't simply returning `self.___call__(*x) - argsi + x` be equivalent?

Reply:
> Wouldn't simply returning `self.___call__(*x) - argsi + x` be equivalent?

No, because `__call__()` has a weird property of returning (Python) lists of coordinate arrays. And `argsi` itself may be a tuple of coordinate arrays depending on the input. Is there anything wrong with these functional forms?

> But more interesting, where is this used in this case?

What do you mean (by _"where is this used in this case"_)?  What does "this" refer to? functional form or returned result?

Reply:
where is f() used in this section?

And are you saying that  for numpy 

OK, you need the casting abilities of the functional form when both arguments are not numpy arrays, basically.

Reply:
> where is f() used in this section?

Thank you! It one of the very first incarnations it was used. Not anymore. I will fix this.